docs/Account/methods.md

```diff
@@ -0,0 +1,284 @@
+
+**Returns**
+
+- address (`Promise<BigNumber>`): A Promise resolving to BigNumber indicating the nonce of the smart account.
+
+### index
+
+This is used to retrieve the index of the current active smart account.
+
+**Usage**
+```jsx
+const index = smartAccount.index;
+```
+
+**Returns**
+- index (`number`): A number indicating the index of current active smart account.
+
+## UserOp Methods
+
+### buildUserOp( )
+This method is used for configuring and setting up all properties of the `userOp` object.  It involves preparing and populating the userOp with necessary information and parameters required for execution by bundler.
+
+**Usage**
+
+For example, in the context of creating a userOp for an `addComment` transaction, an instance of the contract is created, then a basic transaction object is created that holds the necessary address and data from the transaction. Finally, a userOp is created using the Smart Account's `buildUserOp` method⁠.
+
+```jsx
+const contractAddress = "contract address";
+const provider = new ethers.providers.JsonRpcProvider("rpc url");
+
+const blogContract = new ethers.Contract(
+  contractAddress,
+  abi, // contract abi
+  provider,
+);
+
+const createComment =
+  await blogContract.populateTransaction.addComment("comment");
+
+const tx1 = {
+  to: contractAddress,
+  data: createComment.data,
+};
+
+const userOp = await smartAccount.buildUserOp([tx1]);
+```
+
+**Parameters**
+- transactions (`Transaction[]`, required): The required argument  is an array of transactions. You can pass multiple transactions into a userOp if you would like to batch them together into one transaction.
+- buildUseropDto (`BuildUserOpOptions`): One can also pass these options to customize how a userOp is built.
+
+  ```ts
+  type BuildUserOpOptions = {
+    overrides?: Overrides;
+    skipBundlerGasEstimation?: boolean;
+    params?: ModuleInfo;
+    nonceOptions?: NonceOptions; 
+    forceEncodeForBatch?: boolean;
+    paymasterServiceData?: SponsorUserOperationDto;
+  }
+  ```
+  Let's look at each of these params:
+  1. overrides (`Overrides`): one can override any of the values of the userOp when it is being constructed.
+
+      ```ts
+      type BigNumberish = BigNumber | Bytes | bigint | string | number;
+
+      type Overrides = {
+        callGasLimit?: BigNumberish;
+        verificationGasLimit?: BigNumberish;
+        preVerificationGas?: BigNumberish;
+        maxFeePerGas?: BigNumberish;
+        maxPriorityFeePerGas?: BigNumberish;
+        paymasterData?: string;
+        signature?: string; // todo - add the definition
+      }
+      ```
+
+  2. skipBundlerGasEstimation (`boolean`): This boolean param can be set to false to skip some Gas Estimations on the bundler which can help with reducing latency on requests.
+
+  3. params (`ModuleInfo`): One can use this param to pass session validation module parameters. Refer to the tutorial to learn more about the session keys.
```

Comment:
This is useful to generate dummy signature when a complex module is being used which requires more information.